lib/client.js

```diff
@@ -1538,7 +1543,9 @@ function write (client, request) {
     contentLength = null
   }
 
-  if (request.contentLength !== null && request.contentLength !== contentLength) {
+  // https://github.com/nodejs/undici/issues/2046
+  // A user agent may send a Content-Length header with 0 value, this should be allowed.
+  if (shouldSendContentLength(method) && contentLength > 0 && request.contentLength !== null && request.contentLength !== contentLength) {
```

Comment:
`nit`:

Let's store it in some var so we avoid call it twice

Reply:
I think we're ok since one's in `write` and other in `writeH2`!
if all good, no need to respond. thanks!

Reply:
Missed that, thanks!